cmd/dagger/shell_commands.go

```diff
@@ -439,6 +440,38 @@ func (h *shellCallHandler) registerCommands() { //nolint:gocyclo
 Writes any specified operands, separated by single blank (' ') characters and followed by a newline ('\n') character, to the standard output. If the -n option is specified, the trailing newline is suppressed.
 `,
 		},
+		&ShellCommand{
+			Use:  ".printenv [name]",
+			Args: MaximumArgs(1),
+			Description: `Show available environment variables or a specific variable
+
+If no name is provided, all environment variables are printed. If a name is provided, the value of that environment variable is printed.
+			`,
+			State: NoState,
+			Run: func(ctx context.Context, cmd *ShellCommand, args []string, _ *ShellState) error {
+				hc := interp.HandlerCtx(ctx)
+
+				if len(args) == 0 {
+					// Print all environment variables
+					hc.Env.Each(func(name string, v expand.Variable) bool {
+						fmt.Fprintf(hc.Stdout, "%s=%s\n", name, v.String())
+						return true
+					})
```

Comment:
FYI: hc.Env.Each is an iterator, so you can do:

```
for name, vr := range hc.Env {
    fmt.Fprintf(hc.Stdout, "%s=%s\n", name, vr)
}
```

(As a bonus, this skips the explicit call to String, which is implied by %s)

Reply:
Thank you @mvdan, making that change now 🙏